### PR TITLE
Add empty HotspotsTableDataSource

### PR DESCRIPTION
**Pull request by rita-g-sonarsource**
Part of #1795

### CODE REVIEW

**Comment by rita-g-sonarsource** [comment]
@duncanp-sonar this is infrastructure for the table data source; it doesn't handle yet the IssueLocationStore.

File: src/IssueViz.Security/HotspotsList/TableDataSource/HotspotsTableConstants.cs [hunk 1]
```diff
@@ -22,6 +22,9 @@ namespace SonarLint.VisualStudio.IssueVisualization.Security.HotspotsList.TableD
 {
     internal static class HotspotsTableConstants
     {
+        public static readonly string TableDisplayName = Resources.HotspotsToolWindowCaption;
         public const string TableManagerIdentifier = nameof(TableManagerIdentifier);
+        public const string TableIdentifier = nameof(TableIdentifier);
```
Comment (duncanp-sonar) [inline src/IssueViz.Security/HotspotsList/TableDataSource/HotspotsTableConstants.cs:27]: Am I missing something - the identifier for the table is the string "TableIdentifier"? Same for the type identifier.

Those aren't very descriptive names and there's nothing to suggest they relate to SonarLint. I'm guessing those identifiers are ones we're providing to VS so consumers can get hold of the table. If so, then we also want to reduce the possibility of name conflicts.
Either way, I think the identifiers should have some reference to SonarLint.

File: src/IssueViz.Security/IssueViz.Security.csproj [hunk 2]
```diff
@@ -45,6 +45,9 @@
   <ItemGroup Label="Package references">
     <PackageReference Include="Microsoft.VisualStudio.ComponentModelHost" Version="14.0.25424" />
     <PackageReference Include="Microsoft.VisualStudio.Language.Intellisense" Version="14.0.23205" />
+    <PackageReference Include="Microsoft.VisualStudio.Imaging" Version="15.0.26201" />
```
Comment (duncanp-sonar) [inline src/IssueViz.Security/IssueViz.Security.csproj:48]: It looks like this is the first time we've used this package in product code. Why is it version 15?
The only existing ref I could find was in test code to v14.

File: src/IssueViz.Security/IssueViz.Security.csproj [hunk 3]
```diff
@@ -45,6 +45,9 @@
...
+    <PackageReference Include="Microsoft.VisualStudio.ImageCatalog" Version="14.3.25407" />
+    <PackageReference Include="Microsoft.VisualStudio.Imaging.Interop.14.0.DesignTime" Version="16.7.30328.74" />
```
Comment (duncanp-sonar) [inline src/IssueViz.Security/IssueViz.Security.csproj:50]: I see we're already referencing this in IssueViz, but I don't understand why it's version 16.

File: src/IssueViz.Security/HotspotsList/TableDataSource/HotspotsTableDataSource.cs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Shell.TableManager;
+using SonarLint.VisualStudio.Core;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Security.HotspotsList.TableDataSource
+{
+    internal interface IHotspotsTableDataSource : ITableDataSource, IDisposable
```
Comment (duncanp-sonar) [inline src/IssueViz.Security/HotspotsList/TableDataSource/HotspotsTableDataSource.cs:29]: Just an observation: the equivalent `ISonarErrorListDataSource' interface doesn't directly expose the `ITableDataSource`.
It depends what the purpose of this interface is for, but it's not providing any level of abstraction at present.

File: src/IssueViz.Security/HotspotsList/TableDataSource/HotspotTableEntry.cs [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+                case StandardTableColumnDefinitions.DocumentName:
+                    content = originalIssue.FilePath;
+                    break;
+
+                case StandardTableColumnDefinitions.Text:
+                    content = originalIssue.Message;
+                    break;
+
+                case StandardTableColumnDefinitions.Line:
+                    content = originalIssue.StartLine;
```
Comment (duncanp-sonar) [inline src/IssueViz.Security/HotspotsList/TableDataSource/HotspotTableEntry.cs:58]: What's the plan for showing the "live" position as the location is updated by the tagger?

File: src/IssueViz.Security/HotspotsList/TableDataSource/HotspotTableEntry.cs [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+                default:
+                    content = null;
+                    return false;
+            }
+
+            return true;
+        }
+
+        #region ITableEntry
```
Comment (duncanp-sonar) [inline src/IssueViz.Security/HotspotsList/TableDataSource/HotspotTableEntry.cs:73]: The label on this region is a bit misleading - it only contains some of the interface members. Is it to group the "unimplemented" members?

File: src/IssueViz.Security/HotspotsList/TableDataSource/HotspotTableEntry.cs [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+            this.hotspot = hotspot;
+        }
+
+        public object Identity => hotspot;
+
+        public bool TryGetValue(string keyName, out object content)
+        {
+            var originalIssue = hotspot.Issue;
+
+            switch (keyName)
```
Comment (duncanp-sonar) [inline src/IssueViz.Security/HotspotsList/TableDataSource/HotspotTableEntry.cs:43]: Presumably hotspots have a severity too?

Comment (duncanp-sonar) [reply src/IssueViz.Security/HotspotsList/TableDataSource/HotspotTableEntry.cs:43]: Perhaps we should check with the PM(s) which columns they want to see in the list.

**Review by duncanp-sonar — COMMENTED** [review]
A few questions

File: src/IssueViz.Security/HotspotsList/TableDataSource/HotspotTableEntry.cs (hunk 5 above)
Comment (rita-g-sonarsource) [reply src/IssueViz.Security/HotspotsList/TableDataSource/HotspotTableEntry.cs:58]: Good question. I wasn't sure if we would even want to show line/column information, as it's not exactly like the error list (e.g. we also don't show project name). If we take the data from the span, then what should we do if the span is invalid? In the error list we return "null" in GetValue which stops the issue from being rendered, but here we would probably want to keep the item in the list.
Do you think it's worth showing line/column in the first place?

File: src/IssueViz.Security/IssueViz.Security.csproj (hunk 2 above)
Comment (rita-g-sonarsource) [reply src/IssueViz.Security/IssueViz.Security.csproj:48]: My bad, I didn't notice that it was 15. 
I had to import it to reference ImageMoniker in HotspotTableEntry.

File: src/IssueViz.Security/IssueViz.Security.csproj (hunk 3 above)
Comment (rita-g-sonarsource) [reply src/IssueViz.Security/IssueViz.Security.csproj:50]: I'll see what happens if I change it.

File: src/IssueViz.Security/HotspotsList/TableDataSource/HotspotsTableDataSource.cs (hunk 4 above)
Comment (rita-g-sonarsource) [reply src/IssueViz.Security/HotspotsList/TableDataSource/HotspotsTableDataSource.cs:29]: I'll remove it for now

File: src/IssueViz.Security/HotspotsList/TableDataSource/HotspotTableEntry.cs (hunk 6 above)
Comment (rita-g-sonarsource) [reply src/IssueViz.Security/HotspotsList/TableDataSource/HotspotTableEntry.cs:73]: Yep, I'll rename

File: src/IssueViz.Security/HotspotsList/TableDataSource/HotspotTableEntry.cs (hunk 7 above)
Comment (rita-g-sonarsource) [reply src/IssueViz.Security/HotspotsList/TableDataSource/HotspotTableEntry.cs:43]: No idea, do we get that in the API?
I think Alex said that the priority (high/medium/low) is taken from a spreadsheet, so I'm not sure we get that back. I'll ask Damien,

File: src/IssueViz.Security/HotspotsList/TableDataSource/HotspotTableEntry.cs (hunk 5 above)
Comment (duncanp-sonar) [reply src/IssueViz.Security/HotspotsList/TableDataSource/HotspotTableEntry.cs:58]: It's possible that there could be multiple hotspots of the same type in the same file. The user will need something to help disambiguate between them.

Comment (rita-g-sonarsource) [reply src/IssueViz.Security/HotspotsList/TableDataSource/HotspotTableEntry.cs:58]: Ok I'll handle that in a separate PR then, along with Severity column.

**Review by duncanp-sonar — APPROVED** [review]

**Pull request merged**